src/main/java/com/aws/greengrass/ggq/Utils.java

```diff
@@ -0,0 +1,677 @@
+                                break;
+                            case 'u':
+                                sb.append((char) Utils.parseLongChecked(cs, i + 1, i + 5, 16));
+                                i += 4;
+                                break;
+                            default:
+                                sb.append(c);
+                                break;
+                        }
+                    } catch (NumberFormatException t) {
+                        break; // bogus string format: ignore quietly
+                    }
+                } else {
+                    sb.append(c);
+                }
+            }
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Same as deepToString, but output string as JSON encoded, escaping
+     * special characters.
+     *
+     * @param o         object to encode.
+     * @param sb        Appendable object to write the output to.
+     * @param maxLength maximum length of the output string.
+     * @return output length.
+     * @throws IOException if the append fails.
+     */
+    public static int deepToStringQuoted(Object o, Appendable sb, int maxLength) throws IOException {
+        if (o instanceof CharSequence) {
+            sb.append('"');
+            CharSequence s = (CharSequence) o;
+            int l0 = s.length();
+            int len = Math.min(l0, Math.max(5, maxLength - 10));
+            int olen = 2;
+            for (int i = 0; i < len; i++) {
+                char c = s.charAt(i);
+                switch (c) {
+                    case '\n':
+                        sb.append("\\n");
+                        olen += 2;
+                        break;
+                    case '\t':
+                        sb.append("\\t");
+                        olen += 2;
+                        break;
+                    default:
+                        if (c < ' ' || c >= 0xFF || c == 0x7F || c == '"') {
+                            sb.append("\\u");
+                            appendHex(c, 4, sb);
+                            olen += 6;
+                        } else {
+                            sb.append(c);
+                            olen++;
+                        }
+                        break;
+                }
+            }
+            if (len < l0) {
+                sb.append(TRUNCATED_STRING);
+                olen += 3;
+            }
+            sb.append('"');
+            return olen;
+        }
+        return deepToString(o, sb, maxLength);
+    }
+
+    /**
+     * Convert an object to a string representation for human readability.
+     *
+     * @param o         object to convert to string.
+     * @param sb        Appendable to write the string into.
+     * @param maxLength maximum length of the output.
+     * @return actual output length.
+     * @throws IOException if the append fails.
+     */
+    public static int deepToString(Object o, Appendable sb, int maxLength) throws IOException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.